app/facades/obituary_search.rb

```diff
@@ -21,13 +21,52 @@ def recent_obituaries
     end
   end
 
-  def search_advanced_obituaries(name, year)
-    json = ObituaryService.new.get_advanced_obituaries(name, year)
+  def advanced_query(params)
+    params = parse_params(params)
+    if params.keys.include?(:name) && params.keys.include?(:year)
+      json = ObituaryService.new.advanced_search_name_and_year(params[:name], params[:year])
+    elsif params.keys.include?(:name) || params.keys.include?(:year)
+      if params[:name]
+        json = ObituaryService.new.get_obituaries_by_name(params[:name])
+      elsif params[:year]
+        json = ObituaryService.new.advanced_search_year(params[:year])
+      end
+    end
+    if json.present?
+      json.map do |obituary_data|
+        ApiObituary.new(obituary_data)
+      end
+    end
+  end
```

Comment:
this method is definitely beefier than I would like, and features nested iteration, but it is getting the job done for now! If someone wants to pull it down and refactor definitely be my guest. I'm just not sure I have it in me at this point in the project. 😅